### ISSUE TITLE
Payment intents sometimes updated with incorrect application fee

### ISSUE THREAD

**Issue by brianyu28** [Automattic/woocommerce-payments#4679]
### Describe the bug

We've received a couple of reports of instances where a transaction is processed for a U.S. based Stripe account with a U.S. card, but the application fee is 4.4% + $0.30 instead of 2.9% + $0.30 (in other words, there's an additional 1.5% fee added, which we would usually only expect for international cards).

So far, these cases share the following in common:
- Transactions in test mode.
- In the initial Stripe request to create the payment intent, the `application_fee_amount` is the correct value.
- Later, there's a request to update the payment intent, and the `application_fee_amount` gets set to the incorrect value.
- In the order notes that the merchant sees, the fee breakdown _says_ 2.9% + $0.30 but the fee that's shown there is for more than that.

### To Reproduce

We haven't been able to reproduce this, but we've seen the following examples of it happening:

* p1661941605350279-slack-C7U3Y3VMY
* p1659711128533979-slack-CGGCLBN58

### Actual behavior

4.4% + $0.30 fee is charged for non-international transactions.

### Expected behavior

Only 2.9% + $0.30 fee is charged.

### Additional context

p1661957650245259-slack-C02C00HCJ8J

**Comment by kalessil** [comment]
Here we go with the results of the investigation:
- Happens when using Test cards under UPE
- Not happens in dev-mode (wcpay-dev plugin installed), hence IC will not be able to reproduce (because of [this](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/wc-payment-api/class-wc-payments-api-client.php#L346-L349)).

Root cause: both test cards' country (`wcpay_payment_country` hidden field in UPE) populated with non-US country:
- 5555555555554444 - Brasil
- 4242424242424242 - Great Britain

It seems to be misaligned with https://stripe.com/docs/testing#regulatory-cards. 
@brianyu28 can you confirm you have the same countries populated for the test card, please?

**Comment by brianyu28** [comment]
Thanks for looking into it — confirmed that I'm also getting those values for `wcpay_payment_country`, even though Stripe's documentation suggests that `4242424242424242` should be US.

**Comment by kalessil** [comment]
Thanks! Looks like we need to ping Stripe about the country mismatch with what the documentation mentions.

**Comment by csmcneill** [comment]
5555365-zen

**Comment by kalessil** [comment]
I asked clarifications around `4242424242424242` in p1663144462776769-slack-C9976E5MJ.
The stripe docs do not list any country for `5555555555554444`. Hence I omit it from the question.

**Comment by kalessil** [comment]
@csmcneill, perhaps we should update WooCommerce docs ([test cards section](https://woocommerce.com/document/payments/testing/#test-card-numbers)) and specify the card country to avoid further confusion. We could look up the resolved countries if needed. What do you think?

**Comment by csmcneill** [comment]
> @csmcneill, perhaps we should update WooCommerce docs ([test cards section](https://woocommerce.com/document/payments/testing/#test-card-numbers)) and specify the card country to avoid further confusion. We could look up the resolved countries if needed. What do you think?

That may be a good idea. @aheckler what say you?

**Comment by aheckler** [comment]
I was waiting to touch anything until we have an update from Stripe in 5555365-zen. I am CC'ed on that ticket, so as soon as we have an answer from them I can change our docs if needed.

**Comment by RadoslavGeorgiev** [comment]
@aheckler I see there has been activity in the ticket. Is this on your radar?

**Comment by aheckler** [comment]
@RadoslavGeorgiev Looks like Stripe is still working on it per their latest reply:

> Our team is still working on this at this time. For some additional context, there appears to be some discrepancies around the around the BIN of the US test card, causing the fee discrepancy in this case. We have a good lead, we're waiting to hear back from our teams on next steps.

**Comment by tijosh** [comment]
Stripe's update:
> Hi folks, Just wanted to provide an update here. We're still working on investigating this further on our end and while we believe we've narrowed down the root cause in this situation, further time will be needed to remediate. We're requesting input from a few teams who own the processes and card insights for next steps on this. I've also brought in your account team to assist with visibility and insights on our request here. I'll reach back out once we have any further updates here but if anything comes up in the meantime, please let me know.Cheers,Conrad

**Comment by jbordonado** [comment]
No news from Stripe since their last update.
Our HEs sent 2 messages (last one: 1 week ago) to check on this, but still no reply.

**Comment by csmcneill** [comment]
Reached out to Stripe here: p1668655948589889-slack-C9976E5MJ

**Comment by achyuthajoy** [comment]
No response from Stripe. Pinged them asking for an update.

**Comment by htdat** [comment]
👋 @Automattic/pulsar 
I am doing triage for this issue as a part of Platform Porter PcreKM-yM-p2. 
This is one of your product areas Pc2DNy-3z-p2 so I am leaving it here for you. <strike>could you check and add this to your backlog?</strike> 

Edit: you also have a spike issue for it https://github.com/Automattic/woocommerce-payments/issues/5463

**Comment by vbelolapotkov** [comment]
That one seems to be blocked by Stripe. @kalessil I'm adding the label and expect you'll follow with Stripe from time to time on this one.

**Comment by kalessil** [comment]
👍  @vbelolapotkov: adding to our current board as "blocked," so we can monitor relevance after dUPE introduction and see which followups are needed.

**Comment by vbelolapotkov** [comment]
> after dUPE introduction and see which followups are needed.

@kalessil what do you mean by that? If deferred intent UPE, then it's live for a while, but maybe it's something else?

**Comment by kalessil** [comment]
@vbelolapotkov: sorry, short version "we'll re-evaluate". Yes, I'm aware dUPE is live for a while.

**Comment by kalessil** [comment]
Re-evaluated on top of dUPE: US/DE/GB/BR/JP card payment fees are applied properly. Closing the issue.
